### PR TITLE
Define pockets for all integral magazines

### PR DESCRIPTION
**Pull request by ymber**
#### Summary
SUMMARY: Infrastructure "Define pockets for all integral magazines"

#### Purpose of change
Puts data in place to start cutting old code.

#### Describe the solution
Add pocket data to all items with integral magazines. This should obsolete `clip_size` going forward.

#### Testing
Game loads, items work.

### CODE REVIEW

File: data/json/items/gunmod/underbarrel.json [hunk 1]
```diff
@@ -43,7 +43,15 @@
...
     "mod_targets": [ "rifle" ],
-    "gun_data": { "ammo": "shot", "skill": "shotgun", "range": 2, "dispersion": 345, "durability": 10, "clip_size": 2 },
+    "gun_data": {
+      "ammo": "shot",
+      "skill": "shotgun",
+      "range": 2,
+      "dispersion": 345,
+      "durability": 10,
+      "clip_size": 2,
+      "pocket_data": [ { "pocket_type": "MAGAZINE", "ammo_restriction": { "shot": 2 } } ]
```
Comment (KorGgenT) [inline data/json/items/gunmod/underbarrel.json:53]: this won't do anything, for any of the gunmods as is

Comment (ymber) [reply data/json/items/gunmod/underbarrel.json:53]: None of these should do anything currently. Gunmods that define `clip_size` should end up working like anything else with an integral magazine so they need pockets.

Comment (KorGgenT) [reply data/json/items/gunmod/underbarrel.json:53]: i think we should put off adding the json for gunmods until they're read so we can avoid the json reporting unvisited members.

Comment (ymber) [reply data/json/items/gunmod/underbarrel.json]: Moved pocket definitions out of `gun_data`.

**Comment by KorGgenT** [comment]
just needs linting.

**Pull request merged**